File: vBox/MainScreenViewController.m
//
//  MainScreenViewController.m
//  vBox
//
//  Created by Rosbel Sanroman on 11/30/14.
//  Copyright (c) 2014 rosbelSanroman. All rights reserved.
//

#import "MainScreenViewController.h"
#import "MyStyleKit.h"
#import "GoogleMapsViewController.h"

@interface MainScreenViewController ()<GoogleMapsViewControllerDelegate>

@end

@implementation MainScreenViewController{
	BOOL shouldShowNavigationBar;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
	
	if([[NSUserDefaults standardUserDefaults] boolForKey:@"debugMode"])
	{
		[self.debugBluetoothButton setHidden:NO];
		[self.bluetoothTableButton setHidden:NO];
	}
	[self.startDriveButton setBackgroundImage:[MyStyleKit imageOfVBoxButtonWithButtonColor:[MyStyleKit gamebookersBlueColor]] forState:UIControlStateNormal];
	[self.drivingHistoryButton setBackgroundImage:[MyStyleKit imageOfVBoxButtonWithButtonColor:[MyStyleKit myOrange]] forState:UIControlStateNormal];
	
	[self.debugBluetoothButton setBackgroundImage:[MyStyleKit imageOfVBoxButtonWithButtonColor:[MyStyleKit route66Color]] forState:UIControlStateNormal];
	[self.bluetoothTableButton setBackgroundImage:[MyStyleKit imageOfVBoxButtonWithButtonColor:[MyStyleKit route66Color]] forState:UIControlStateNormal];
	
	shouldShowNavigationBar = YES;
}

-(void)viewWillAppear:(BOOL)animated
{
	[super viewWillAppear:animated];
//	[[UIApplication sharedApplication] setStatusBarStyle:UIStatusBarStyleLightContent];
	[self.navigationController setNavigationBarHidden:YES animated:YES];
}

-(UIStatusBarStyle)preferredStatusBarStyle
{
    return UIStatusBarStyleLightContent;
}

- (void)viewWillDisappear:(BOOL)animated
{
	[super viewWillDisappear:animated];
	
	if(shouldShowNavigationBar)
		[self.navigationController setNavigationBarHidden:NO animated:YES];
	
	shouldShowNavigationBar = YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
	// Get the new view controller using [segue destinationViewController].
	// Pass the selected object to the new view controller.
	if([segue.identifier isEqualToString:@"googleMapsSegue"])
	{
		shouldShowNavigationBar = NO;
		GoogleMapsViewController *googleMaps = segue.destinationViewController;
		googleMaps.delegate = self;
	}
}

#pragma mark - GoogleMapsViewControllerDelegate Methods

-(void)didTapStopRecordingButton
{
	[self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: vBox/ViewControllers/MainScreenViewController.swift
//
//  MainScreenViewController.swift
//  vBox
//
//  Swift implementation of main screen navigation
//

import UIKit

// MARK: - Main Screen View Controller

final class MainScreenViewControllerSwift: UIViewController {

    // MARK: - IBOutlets

    @IBOutlet private weak var startDriveButton: UIButton!
    @IBOutlet private weak var drivingHistoryButton: UIButton!
    @IBOutlet private weak var debugBluetoothButton: UIButton!
    @IBOutlet private weak var bluetoothTableButton: UIButton!

    // MARK: - Properties

    private var shouldShowNavigationBar = true

    // MARK: - Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupButtons()
        configureDebugMode()
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationController?.setNavigationBarHidden(true, animated: true)
    }

    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)

        if shouldShowNavigationBar {
            navigationController?.setNavigationBarHidden(false, animated: true)
        }
        shouldShowNavigationBar = true
    }

    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }

    // MARK: - Setup

    private func setupButtons() {
        // Configure button styles using MyStyleKit
        let blueImage = MyStyleKit.image(ofVBoxButtonWithButtonColor: MyStyleKit.gamebookersBlueColor())
        let orangeImage = MyStyleKit.image(ofVBoxButtonWithButtonColor: MyStyleKit.myOrange())
        let debugImage = MyStyleKit.image(ofVBoxButtonWithButtonColor: MyStyleKit.route66Color())

        startDriveButton.setBackgroundImage(blueImage, for: .normal)
        drivingHistoryButton.setBackgroundImage(orangeImage, for: .normal)
        debugBluetoothButton.setBackgroundImage(debugImage, for: .normal)
        bluetoothTableButton.setBackgroundImage(debugImage, for: .normal)
    }

    private func configureDebugMode() {
        let isDebugMode = UserDefaults.standard.bool(forKey: "debugMode")
        debugBluetoothButton.isHidden = !isDebugMode
        bluetoothTableButton.isHidden = !isDebugMode
    }

    // MARK: - Navigation

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "googleMapsSegue" {
            shouldShowNavigationBar = false

            if let googleMapsVC = segue.destination as? GoogleMapsViewController {
                googleMapsVC.delegate = self
            }
        }
    }
}

// MARK: - GoogleMapsViewControllerDelegate

extension MainScreenViewControllerSwift: GoogleMapsViewControllerDelegate {
    func didTapStopRecordingButton() {
        dismiss(animated: true, completion: nil)
    }
}
